docs/attributes.md

```diff
@@ -33,7 +33,8 @@ Configurable (with defaults)
 * `node['splunk']['config']['authentication']` - Data bag item used to configure authentication (`nil` - authentication not managed by chef)
 * `node['splunk']['config']['host']` - Hostname to configure the Splunk instance to report as. (EC2 Instance ID or Fully Qualified Domain Name)
 * `node['splunk']['config']['roles']` - Data bag item used to configure roles (`nil` - roles not managed by chef)
-* `node['splunk']['config']['secrets']` - Coordinate String (see [data bags documentation][data_bags]), pointing to a key within a Chef Vault encrypted data bag item used to configure the splunk.secret file. The value must be a string. (`nil` - secrets not managed by chef).  Note: this is currently not supported on windows.
+* `node['splunk']['config']['secrets']` - A Contextual Hash of coordinate strings (see [data bags documentation][data_bags]), pointing to a key within a data bag item used to configure the splunk.secret file. (`nil` - secrets not managed by chef).  Note: this is currently not supported on windows.
+* `node['splunk']['config']['admin_password']` - Coordinate String (see [data bags documentation][data_bags]), pointing to a key within a data bag item used to configure the local admin password. The value must be a string. (`nil` - password is randomly generated and changed on every chef-client run).
```

Comment:
The question is more of, would we set it environment wide, and would there be nodes in the environment that would not be allowed to decrypt the data bag (either without the same shared secret or not being part of the chef-vault), and/or would we want to set it environment wide and have it vary this by node type? (e.g. let the search head have one password, and indexer slaves have another)

If we expect so, then we should go contextual hash... but if we feel like this would be set on a per-node basis, or could be set once globally for all nodes in an environment, then no problem.

With the secrets, as we initially didn't support changing secrets we were going to have some nodes that we wouldn't manage and some that we would within the environment, hence the coordinate hash. And instead of jockeying with different node/role+environment based settings, made this a contextual hash so we could set it once in the environment and be done.

Reply:
I went ahead and converted it over to a contextual hash in https://github.com/cerner/cerner_splunk/pull/162/commits/9c48aba91b1287d58f0a34685907d3384453662d.